torchaudio/datasets/tedlium.py

```diff
@@ -0,0 +1,133 @@
+import os
+from typing import Tuple
+
+import torchaudio
+from torch import Tensor
+from torch.utils.data import Dataset
+from torchaudio.datasets.utils import (
+    download_url,
+    extract_archive,
+    walk_files,
+)
+
+RELEASE = "release1"  # Default release
+
+_RELEASE_CONFIGS = {
+    "release1": {
+        "folder_in_archive": "TEDLIUM_release1",
+        "url": "http://www.openslr.org/resources/7/TEDLIUM_release1.tar.gz",
+        "checksum": "ffd31f96d81a21bf4928eaf9bb0b0c2dea7a5247",
+        "data_path": "",
+        "subset": "train",
+    },
+    "release2": {
+        "folder_in_archive": "TEDLIUM_release2",
+        "url": "http://www.openslr.org/resources/19/TEDLIUM_release2.tar.gz",
+        "checksum": "5c8fb045246d1c64296f57b47aa7dc79d16b184f",
+        "data_path": "",
+        "subset": "train",
+    },
+    "release3": {
+        "folder_in_archive": "TEDLIUM_release-3",
+        "url": "http://www.openslr.org/resources/51/TEDLIUM_release-3.tgz",
+        "checksum": "685d27c39c53217383d7933cc405a07048004127",
+        "data_path": "data/",
+        "subset": None,
+    },
+}
+
+
+def load_tedlium_item(
+    fileid: str, line: int, path: str, ext_audio: str, ext_txt: str
+) -> Tuple[Tensor, int, str, int, int, int]:
+    transcript_path = os.path.join(path, "stm/", fileid)
+    with open(transcript_path + ext_txt) as f:
+        transcript = f.readlines()[line]
+        talk_id, _, speaker_id, start_time, end_time, identifier, transcript = transcript.split(
+            " ", 6
+        )
+
+    wave_path = os.path.join(path, "sph/", fileid)
+    waveform, sample_rate = torchaudio.load(wave_path + ext_audio)
+    print(wave_path + ext_audio)
+    # Calculate indexes for start time and endtime
+    start_time = int(float(start_time) * sample_rate)
+    end_time = int(float(end_time) * sample_rate)
+    print(start_time, end_time)
+    waveform = waveform[:, start_time:end_time]
+    return (
+        waveform,
+        sample_rate,
+        transcript,
+        talk_id,
+        speaker_id,
+        identifier,
+        transcript,
+    )
+
+
+class TEDLIUM(Dataset):
+    """
+    Create a Dataset for Tedlium. Each item is a tuple of the form:
+    waveform, sample_rate, utterance, speaker_id, chapter_id, utterance_id
+    """
+
+    _ext_txt = ".stm"
+    _ext_audio = ".sph"
+    _folder_audio = "sph/"
```

Comment:
I used them to mimic structure on the Librispeech dataset:https://github.com/pytorch/audio/blob/c692fe99ac353f0e2fc7b5ad7162f037a70a35fe/torchaudio/datasets/librispeech.py#L75-L76

I agree there is no benefit from having them as a variable rather than a attribute but there is the consistency also if we want to keep them. I have no problem replacing them with hardcoding but wanted to point this before making the change

Reply:
Yes, makes sense